generators/heroku/index.js

```diff
@@ -31,31 +31,49 @@ module.exports = HerokuGenerator.extend({
         this.angularAppName = this.getAngularAppName();
         this.buildTool = this.config.get('buildTool');
         this.applicationType = this.config.get('applicationType');
+        this.herokuAppName = this.config.get('herokuAppName');
     },
 
     prompting: function () {
         var done = this.async();
 
-        var prompts = [
-            {
-                type: 'input',
-                name: 'herokuDeployedName',
-                message: 'Name to deploy as:',
-                default: this.baseName
-            },
-            {
-                type: 'list',
-                name: 'herokuRegion',
-                message: 'On which region do you want to deploy ?',
-                choices: ['us', 'eu'],
-                default: 0
-            }];
-
-        this.prompt(prompts).then(function (props) {
-            this.herokuDeployedName = _.kebabCase(props.herokuDeployedName);
-            this.herokuRegion = props.herokuRegion;
-            done();
-        }.bind(this));
+        if (this.herokuAppName && this.herokuAppName.length != 0) {
+            exec('heroku apps:info --json', function (err, stdout) {
+                if (err) {
+                    this.abort = true;
+                    this.log.error(err);
+                } else {
+                    var json = JSON.parse(stdout);
+                    this.herokuAppName = json['app']['name'];
```

Comment:
Good question. The value in `json['app']['name']` is the source of truth if there happens to be a conflict between what's in `.yo-rc` and what's returned by `heroku apps:info`. The only way there would be conflict is if the user changed the Heroku app outside of the `yo jhipster:heroku` workflow, which is unlikely.

Thus, the `this.config.get('herokuAppName')` property could be simplified to a boolean `this.config.get('herokuAppExists')`. But think it might prove useful in the future to have the app name in `.yo-rc` (it certainly doesn't hurt).

Makes sense?